fix(text-field): Add support for <textarea> (#118)

bmihelac opened the pull request:
Please do not merge.

This is idea for adding support for <textarea>. What do you think?

moog16 commented:
@bmihelac is that really all the changes we need? To either switch out `input` or `textarea`? I thought it would be more complex than this lol

bmihelac commented:
It looks like it is. Would appreciate review to assure something important is not missing.

bmihelac commented:
@bonniezhou thanks for checking! will add tests soon

bmihelac commented:
Added tests and updated docs. Please let me know if anything else is needed.

bonniezhou commented on test/screenshot/text-field/index.js:
```diff
@@ -90,10 +109,31 @@ const textFields = variants.map((variant) => {
+        return helperTextMap.map((helperText) => {
+          const props = Object.assign({}, variant, dense, disabled, helperText, isRtl);
+          const key = JSON.stringify(props);
+          return <TestTextareaField {...props} key={key} id={key} />;
```
I think you can just use `<TestField textarea ...>` here, no need to create `TestTextareaField` :)

bmihelac commented:
Thanks for review @bonniezhou! I would update code as requested.

One thing I am not sure about is if I would be able to update screenshot tests as I do not have `MDC_GCLOUD_SERVICE_ACCOUNT_KEY`(as specified here:
https://github.com/material-components/material-components-web-react/blob/master/docs/screenshot-tests.md#step-2-run-a-docker-container). Any pointer for this?

bmihelac commented:
Changes has been made (except screenshots that is commented above).
Please let me know if anything else is needed.

moog16 commented on packages/text-field/Input.js:
```diff
@@ -121,6 +121,7 @@ export default class Input extends React.Component {
 
   render() {
     const {
+      inputType,
```
@bmihelac do you mind changing this from `inputType` to `tag`? I saw JamesFriedman from RMWC is using this prop name, and I like it! 

Also using it in drawer, so it will be consistent with the rest of the components that we will build.

moog16 commented:
I made a list:

- I created a PR for the tests to run: https://github.com/material-components/material-components-web-react/pull/328
- @bmihelac I also created the golden...could you update it in your branch?
https://github.com/material-components/material-components-web-react/pull/328/files#diff-fa04b5568411785c79cb068863c6e41a found in the `test/screenshot/golden.json`
- Add this change https://github.com/material-components/material-components-web-react/pull/328/files#diff-c645555be4bdfcdad7e2152c50ccc44c found in `test/screenshot/screenshot.js`, this is required because puppeteer didn't have enough memory to run the screenshot test page. See comment in code. 
- please sign the PR

Thanks again for all your help!

bmihelac commented:
updated golden, screenshot test, rebased and signed.

@moog16 I am not sure if request change for renaming `inputType` to `tag` is still relevant - please let me know.

moog16 commented:
@bmihelac no - I actually deleted that comment about 2 minutes after. Sorry for the confusion!

moog16 commented:
@bmihelac looks like you have a conflict. Please fix. and I'll rerun tests on travis

bmihelac commented:
Hi @moog16. The conflict github is mentioning is regarded to golden. I have copied it from your comment https://github.com/material-components/material-components-web-react/pull/266#issuecomment-428015427 and believe that without it screenshot tests would fail. Can you check?

moog16 commented:
Yup just fixed!

moog16 commented:
closed #228 for https://github.com/material-components/material-components-web-react/pull/330

moog16 commented:
@bmihelac I cla/google still not passing. I think one of your commits actually has to say "I signed it"

bmihelac commented:
I have signed "Google Individual CLA" and I can confirm it here:
https://cla.developers.google.com/clas

bmihelac commented:
I signed it!

bmihelac commented:
> I think one of your commits actually has to say "I signed it"

it looks comment here should say "I have signed" or "I signed it". That was what triggered googlebot to update status

moog16 commented:
Thanks! - I think "I signed it"
